src/convenience/keyboard.ts

```diff
@@ -501,18 +502,19 @@ export class Keyboard {
      * You can use the static button builder methods to create keyboard button
      * objects.
      *
-     * @param buttons A two-dimensional button array
+     * @param source A two-dimensional button array
      * @param options Optional options for the custom keyboard
      */
     static from(
-        buttons: (string | KeyboardButton)[][],
+        source: (string | KeyboardButton)[][] | Keyboard,
         options: Omit<ReplyKeyboardMarkup, "keyboard"> = {},
     ): Keyboard {
+        if (source instanceof Keyboard) return source;
```

Comment:
well this part is not always correct. what if we wanted to override options from another keyboard?

`const k2 = Keyboard.from(k1, {...})`

Reply:
Good catch, will fix